cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/CoreSchedulerService.java

```diff
@@ -111,21 +121,77 @@ public void addSchedules(final Iterable<? extends ProgramSchedule> schedules) th
       @Override
       public Void run(ProgramScheduleStoreDataset store) throws AlreadyExistsException {
         store.addSchedules(schedules);
+        for (ProgramSchedule schedule : schedules) {
+          try {
+            scheduler.addProgramSchedule(schedule);
+          } catch (SchedulerException e) {
+            // TODO: temporarily catch the SchedulerException and throw AlreadyExistsException.
```

Comment:
This is in a transaction, so throwing an exception here will prevent adding the ProgramSchedule